bin/improver-wxcode

```diff
@@ -0,0 +1,96 @@
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""CLI to generate weather symbols."""
+
+import argparse
+import iris
+import numpy as np
+from argparse import RawTextHelpFormatter
+
+from improver.wxcode.weather_symbols import WeatherSymbols
+from improver.wxcode.wxcode_utilities import expand_nested_lists
+from improver.wxcode.wxcode_decision_tree import wxcode_decision_tree
+
+
+def main():
+    """Parser to accept input data and an output destination before invoking
+    the weather symbols plugin.
+    """
+
+    # Get current weather symbol decision tree and populate a list of
+    # required inputs for printing.
+    queries = wxcode_decision_tree()
+    requirements = {}
+
+    for query in queries.itervalues():
+        diagnostics = expand_nested_lists(query, 'diagnostic_fields')
+        for index, diagnostic in enumerate(diagnostics):
+            if diagnostic not in requirements:
+                requirements[diagnostic] = []
+            requirements[diagnostic].extend(
+                expand_nested_lists(query, 'diagnostic_thresholds')[index])
+
+    output = []
+    for requirement in requirements:
+        entries = np.array([entry for entry in requirements[requirement]])
+        _, thresholds = np.unique(np.array([item.points.item()
+                                            for item in entries]),
+                                  return_index=True)
+        output.append('{}; thresholds: {}'.format(
+                requirement, ', '.join([
+                        '{} ({})'.format(str(threshold.points.item()),
+                                         str(threshold.units))
+                        for threshold in entries[thresholds]])))
+    n_files = len(output)
+    dlist = (' - {}\n'*n_files)
+
+    parser = argparse.ArgumentParser(
+        description='Calculate a cube of weather symbol codes.\nThis plugin '
+        'requires a specific set of input cubes:\n' +
+        dlist.format(*output),
+        formatter_class=RawTextHelpFormatter)
+
+    parser.add_argument(
+        'input_filepaths', metavar='INPUT_FILES', nargs=n_files,
```

Comment:
Maybe change 'INPUT_FILES' to 'INPUT_FILE' as each argument refers to only one file?

Reply:
It's a bit of a toss up, because it will read incorrectly one way or another. The improver wxcode -h gives a list of positional arguments where the plural makes more sense:
```
positional arguments:
  INPUT_FILES  Paths to files containing the required input cubes.
  OUTPUT_FILE  The output path for the processed NetCDF.
````

I think I will leave it so this statement reads correctly.